python/ml4ir/base/data/ranklib_to_ml4ir.py

```diff
@@ -0,0 +1,60 @@
+import pandas as pd
+import argparse
+import logging
+
+#Constanst for the column names given to the query id and graded relevance.
+QUERY_ID_NAME = 'qid'
+RELEVANCE_NAME = 'relevance'
+
+
+def process_line(line, keep_additional_info):
+    if keep_additional_info == 1:
+        feature_values = line.replace('#', '').replace(' = ', ':').strip().split()
+    else:
+        feature_values = line.split('#')[0].strip().split()
+    feature_values[0] = RELEVANCE_NAME + ':' + feature_values[0]
+    r = {}
+    for fv in feature_values:
+        feat = fv.split(':')[0].strip()
+        val = fv.split(':')[1].strip()
+        if feat == RELEVANCE_NAME or feat == QUERY_ID_NAME:
```

Comment:
@mohazahran Does using this hardcoded keys, limit us to only using ranklib? Would this work even if we want to use ml4ir for libsvm format in general? ml4ir could be used for other ML tasks as well.

Reply:
I believe libsvm uses ranklib format as well which specifies the features as "name:value" so it shouldn't be limiting.
However, I noticed some datasets as "Yahoo LETOR" stores only non zero features (for space efficiency). I will handle that as well

Reply:
Right. I just wanted us to be able to use this script to convert data that is not necessarily limited to ranking(ranklib) alone. So I was wondering if using `query_id` and `relevance` was a good idea or not. May be we can have two paths based on an argument to specify if the dataset is using ranklib? A better design would be creating libsvm parser and then ranklib parser which inherits the libsvm parser. That way most of the functionality can be shared.

Reply:
I'm not sure that I fully understand your point. You see, ranklib and libsvm shares the same format (with minor differences). Both require a query Id and a target relevance present. So explicitly using both should not be limiting the script.